Modules/_curses_panel.c

```diff
@@ -622,57 +641,58 @@ static PyMethodDef PyCurses_methods[] = {
 };
 
 /* Initialization function for the module */
-
-
-static struct PyModuleDef _curses_panelmodule = {
-        PyModuleDef_HEAD_INIT,
-        "_curses_panel",
-        NULL,
-        sizeof(_curses_panelstate),
-        PyCurses_methods,
-        NULL,
-        _curses_panel_traverse,
-        _curses_panel_clear,
-        _curses_panel_free
-};
-
-PyMODINIT_FUNC
-PyInit__curses_panel(void)
+static int
+_curses_exec(PyObject *m)
 {
-    PyObject *m, *d, *v;
-
-    /* Create the module and add the functions */
-    m = PyModule_Create(&_curses_panelmodule);
-    if (m == NULL)
-        goto fail;
-    d = PyModule_GetDict(m);
-
+    _curses_panelstate *state = get_curses_panelstate(m);
     /* Initialize object type */
-    v = PyType_FromSpec(&PyCursesPanel_Type_spec);
-    if (v == NULL)
-        goto fail;
-    ((PyTypeObject *)v)->tp_new = NULL;
-    get_curses_panelstate(m)->PyCursesPanel_Type = v;
+    state->PyCursesPanel_Type = (PyTypeObject *)PyType_FromModuleAndSpec(
+        m, &PyCursesPanel_Type_spec, NULL);
+    if (state->PyCursesPanel_Type == NULL) {
+        return -1;
+    }
+
+    if (PyModule_AddType(m, state->PyCursesPanel_Type) < 0) {
+        return -1;
+    }
 
     import_curses();
     if (PyErr_Occurred())
-        goto fail;
+        return -1;
 
+    PyObject *d = PyModule_GetDict(m);
     /* For exception _curses_panel.error */
-    get_curses_panelstate(m)->PyCursesError = PyErr_NewException("_curses_panel.error", NULL, NULL);
-    PyDict_SetItemString(d, "error", get_curses_panelstate(m)->PyCursesError);
+    state->PyCursesError = PyErr_NewException(
+        "_curses_panel.error", NULL, NULL);
+    PyDict_SetItemString(d, "error", state->PyCursesError);
 
     /* Make the version available */
-    v = PyUnicode_FromString(PyCursesVersion);
+    PyObject *v = PyUnicode_FromString(PyCursesVersion);
     PyDict_SetItemString(d, "version", v);
```

Comment:
Can you please check for PyDict_SetItemString() error? (You need to call `Py_DECREF(v);` in the error path.)